src/Nethermind/Nethermind.Evm/CodeAnalysis/ByteCodeValidator.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Nethermind.Logging;
+using Nethermind.Core.Attributes;
+using Nethermind.Core.Specs;
+using Org.BouncyCastle.Crypto.Agreement.Srp;
+using System.Diagnostics.CodeAnalysis;
+using Nethermind.Evm.EOF;
+
+namespace Nethermind.Evm.CodeAnalysis
+{
+    internal class ByteCodeValidator
+    {
+
+        private readonly EvmObjectFormat? EofFormatChecker;
+        public ByteCodeValidator(ILogManager logManager = null)
+            => EofFormatChecker = new EvmObjectFormat(logManager);
+
+        public bool HasEofPrefix(ReadOnlySpan<byte> code) => EofFormatChecker.IsEof(code);
+        public bool ValidateBytecode(ReadOnlySpan<byte> code, IReleaseSpec spec,
+            [NotNullWhen(true)] out EofHeader? header)
+        {
+            header = null;
+            return !spec.IsEip3540Enabled || !EofFormatChecker.IsEof(code)
+                ? CodeDepositHandler.CodeIsValid(spec, code.ToArray())
+                : EofFormatChecker.TryExtractEofHeader(code, out header);
```

Comment:
`EofFormatChecker.TryExtractEofHeader` will code call `EofFormatChecker.ValidateBody` thus not finishing full EOF validation, I think it should call `EofFormatChecker.IsValidEof`

Reply:
that is for 3670 though not for this one.